beacon-chain/p2p/peers/status.go

```diff
@@ -429,6 +429,9 @@ func (p *Status) Decay() {
 // Returns the best finalized root, epoch number, and list of peers that are at or beyond that epoch.
 func (p *Status) BestFinalized(maxPeers int, ourFinalizedEpoch uint64) ([]byte, uint64, []peer.ID) {
 	connected := p.Connected()
+	if len(connected) == 0 {
+		return nil, 0, nil
```

Comment:
dont think this is correct, they wanted us to avoid returning a nil/bad root here which was the main concern around having 0 peers. This would cause our node to panic if we process a nil root.